nifi-nar-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketClient.java

```diff
@@ -81,6 +87,16 @@
             .defaultValue("3 sec")
             .build();
 
+    public static final PropertyDescriptor SESSION_MAINTENANCE_INTERVAL = new PropertyDescriptor.Builder()
+            .name("session-maintenance-interval")
+            .displayName("Session Maintenance Interval")
+            .description("The interval between session maintenance activities.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("10 sec")
```

Comment:
The CONNECTION_TIMEOUT is a timeout for the initial connection attempt and idle connection timeout is different (IIRC 5 min by default and not configurable from controller ATM, probably we should it configurable though). The maintenance does not take any effect before a connection is established.